provisioner/interface.go

```diff
@@ -0,0 +1,17 @@
+package provisioner
+
+import (
+	"github.com/yard-turkey/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	"github.com/yard-turkey/lib-bucket-provisioner/provisioner/auth"
+)
+
+// Provisioner the interface to be implemented by users of this
+// library and executed by the Reconciler
+type Provisioner interface {
+	// Provision should be implemented to handle bucket creation
+	// for the target object store
+	Provision(*v1alpha1.ObjectBucketClaim) (*v1alpha1.ObjectBucket, *auth.S3AccessKeys, error)
+	// Delete should be implemented to handle bucket deletion
+	// for the target object store
+	Delete(claim *v1alpha1.ObjectBucketClaim) error
```

Comment:
do you think we may need to pass the reclaimPolicy as another arg in `Delete()`?  Or, add it to the OBC...

Reply:
This actually needs to be changed to accept an OB, not an OBC.